astronomer/providers/amazon/aws/operators/redshift_data.py

```diff
@@ -43,6 +43,25 @@ def execute(self, context: Context) -> None:  # type: ignore[override]
         if response.get("status") == "error":
             self.execute_complete(context, event=response)
         context["ti"].xcom_push(key="return_value", value=query_ids)
+
+        completed_id = []
+        for qid in query_ids:
+            resp = redshift_data_hook.conn.describe_statement(Id=qid)
+            status = resp["Status"]
+            if status == "FAILED":
+                err_msg = f"Error: {resp['QueryString']} query Failed due to {resp['Error']}"
+                msg = f"context: {context}, error message: {err_msg}"
+                raise AirflowException(msg)
+            elif status == "ABORTED":
+                err_msg = "The query run was stopped by the user."
+                msg = f"context: {context}, error message: {err_msg}"
+                raise AirflowException(msg)
+            elif status == "FINISHED":
+                completed_id.append(qid)
```

Comment:
Instead of looping over all the queries to check whether they are finished, would it make sense to break the loop and enter deferral as soon as we find a query that is still running(or not in any of the terminal states)?

Reply:
Indeed. Let me rewrite it. Thanks for the great suggestion!